src/conventions/h-group/clue-interpretation/focus-possible.js

```diff
@@ -72,14 +74,25 @@ function find_colour_focus(game, suitIndex, action) {
 			}
 
 			// Even if a finesse is possible, it might not be a finesse
-			focus_possible.push({ suitIndex, rank: next_rank, save: false, connections: Utils.objClone(connections), interp: CLUE_INTERP.PLAY });
+			promised.push(focused_card);
+			const possible_connections = resolve_bluff(game, giver, target, Utils.objClone(connections), promised, { suitIndex, rank: next_rank });
```

Comment:
I'd prefer not to clone `connections` unnecessarily, since that can be expensive. It looks like the only modifications to `connections` are near the end of `resolve_bluff()`, where either the first connection has its `bluff` property changed to `false` or a few connections are spliced from the array. For the former I would `slice()` the array, clone the first connection and replace that, and for the latter you can use `connections.toSpliced()` instead.